src/validators/validation-helpers.ts

```diff
@@ -1,59 +1,178 @@
+      .withMessage(`${this.beautifyParam(paramName)} must not be empty`)
       .bail();
   }
 
-  static validateInteger(paramName: string, isRequired: boolean = false): ValidationChain {
-    const checker = this.validateRequired(body(paramName), paramName, isRequired);
+  static validateInteger(
+    paramName: string,
+    target: ValidationTarget = ValidationTarget.Body,
+    isRequired: boolean = false
+  ): ValidationChain {
+    const checker = this.validateRequired(
+      this.getChecker(target, paramName),
+      paramName,
+      isRequired
+    );
     return checker
       .not()
       .isArray()
-      .withMessage(`${paramName} must not be an array`)
+      .withMessage(`${this.beautifyParam(paramName)} must not be an array`)
       .bail()
       .isInt()
-      .withMessage(`${paramName} must be an integer`)
+      .withMessage(`${this.beautifyParam(paramName)} must be an integer`)
       .bail()
       .toInt();
   }
 
-  static validateDecimal(paramName: string, isRequired: boolean = false): ValidationChain {
-    const checker = this.validateRequired(body(paramName), paramName, isRequired);
+  static validateDecimal(
+    paramName: string,
+    target: ValidationTarget = ValidationTarget.Body,
+    isRequired: boolean = false
+  ): ValidationChain {
+    const checker = this.validateRequired(
+      this.getChecker(target, paramName),
+      paramName,
+      isRequired
+    );
     return checker
       .not()
       .isArray()
-      .withMessage(`${paramName} must not be an array`)
+      .withMessage(`${this.beautifyParam(paramName)} must not be an array`)
       .bail()
       .isDecimal()
-      .withMessage(`${paramName} must be a decimal`)
+      .withMessage(`${this.beautifyParam(paramName)} must be a decimal`)
       .bail()
       .toFloat();
   }
 
-  static validateDate(paramName: string, isRequired: boolean = false): ValidationChain {
-    const checker = this.validateRequired(body(paramName), paramName, isRequired);
-    return checker.isISO8601({ strict: true }).withMessage(`${paramName} must be a valid Date`).bail();
+  static validateDate(
+    paramName: string,
+    target: ValidationTarget = ValidationTarget.Body,
+    isRequired: boolean = false
+  ): ValidationChain {
+    const checker = this.validateRequired(
+      this.getChecker(target, paramName),
+      paramName,
+      isRequired
+    );
+    return checker
+      .isISO8601({ strict: true })
+      .withMessage(`${this.beautifyParam(paramName)} must be a valid Date`)
+      .bail()
+      .toDate();
+  }
+
+  static validateArray(
+    paramName: string,
+    target: ValidationTarget = ValidationTarget.Body,
+    isRequired: boolean = false
+  ): ValidationChain {
+    const checker = this.validateRequired(
+      this.getChecker(target, paramName),
+      paramName,
+      isRequired
+    );
+    return checker
+      .isArray()
+      .withMessage(`${this.beautifyParam(paramName)} must be an array`)
```

Comment:
If `paramName` is empty, we can instead put `Body` or `Query` in the message here. This will allow us to cover validation case of the next method that only validates that `Body` is array.